lib/commands/connect/restart.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+var api = require('./shared.js');
+let cli = require('heroku-cli-util');
+
+module.exports = {
+    topic: 'connect',
+    command: 'restart',
+    description: 'Restart a connection',
+    help: 'Clears errors and attempts to resume sync operations',
+    flags: [
+      {name: 'resource', char: 'r', description: 'specific connection resource name', hasValue: true}
+    ],
+    needsApp: true,
+    needsAuth: true,
+    run: cli.command(function (context, heroku) {
+      api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku).then(function(connections){
+          if (connections.length > 1) {
+            console.log("Multiple connections found. Please use '-r' to specify a single connection by resource name.");
+            api.connection_info(connections);
+          } else {
+            connections.forEach(function(connection) {
+              api.request(context.auth.password, 'POST', '/api/v3/connections/' + connection.id + '/actions/restart')
+                .then(function(response) {
+                  switch(response.statusCode) {
+                    case 202:
+                      console.log(api.connection_string(connection) + ' restarted');
+                    break;
+                    case 409:
+                      console.log(response.json['message']);
```

Comment:
I'm not so sure about this one. I feel like we should be just parroting out whatever we get from the API, unless (in the case of the 202) we don't get anything back at all. So if there's an improvement to the wording, I'd rather make that change in the API, so all clients can take advantage of it, rather than expecting each client to write up their own messaging for each status code. Granted, our API errors for things like this are pretty terrible, and should definitely be improved, but that's outside the scope of this PR.